File: 1456-maximum-number-of-vowels-in-a-substring-of-given-length/1456-maximum-number-of-vowels-in-a-substring-of-given-length.js
/**
 * @param {string} s
 * @param {number} k
 * @return {number}
 */
var maxVowels = function(s, k) {
    let idx=0
    let count=0
    let max=0
    let vowels=new Set(["a","e","i","o","u"])
    while(idx+k<=s.length){
        if(idx===0){
            for(let i=0;i<k;i++)
                if(vowels.has(s[i]))
                    count++;
            max=count
        }
        else{
//             console.log(idx, idx+k)
            
//             console.log("to be removed: " + s[idx-1])
//             console.log("to be added: " + s[idx+k-1])
//             console.log(s.slice(idx,idx+k))
            if(vowels.has(s[idx-1])) 
                count--
            if(vowels.has(s[idx+k-1]))
                count++
            
            max=Math.max(max, count)
        }
        
        // console.log("count: "+count)
        idx++
    }
    return max
};

File: 1456-maximum-number-of-vowels-in-a-substring-of-given-length/1456-maximum-number-of-vowels-in-a-substring-of-given-length.py
class Solution:
    def maxVowels(self, s: str, k: int) -> int:
        def isVowel(c):
            return 1 if c in {"a","e","i","o","u"} else 0
            
            
        idx=0
        count=0
        res=0
        
        while idx+k<=len(s):
            if(idx==0):
                for i in range(0, k):
                    count+= isVowel(s[i])
                        
                res=count
            
            else: 
                count-=isVowel(s[idx-1])
                count+=isVowel(s[idx+k-1])
                
            
            res=max(res, count)
            idx+=1
            # print("count: "+count)
        
    
        return res
        